lib/topological_inventory/providers/common/operations/source.rb

```diff
@@ -153,8 +153,9 @@ def update_application(status)
 
         def endpoint
           @endpoint ||= api_client.fetch_default_endpoint(source_id)
-        rescue e
+        rescue => e
           logger.availability_check("Failed to fetch Endpoint for Source #{source_id}: #{e.message}", :error)
+          false # defaults to true because that is what availability_check returns
```

Comment:
Will this affect how the UI shows an availability status?  This seems like a hard 'false' if the `@endpoint` isn't set, which seems correct to me, but I wonder if the UI is doing the right thing here as well?

Reply:
The main problem is that the code messes up since we're checking for `endpoint || application` most places - I could check and see if they're not instances of `TrueClass` I guess as an alternative approach. Should I just do that? Make sure that the values are there and they are the right types?

Reply:
somewhere the endpoint is checked to nil, not to false, what about returning nil?

Reply:
Sure, updated it to return nil since that is what we usually check for.